### PR TITLE
#3606 - Detrack new action

### LINKED ISSUE #3606: Detrack
### App
Detrack

### Actions
* Create Job

### PR DESCRIPTION
**Pull request by GTFalcao**
# New Action
- **Create Job**

Closes #3606 

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4503"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible

File: components/detrack/actions/create-job/create-job.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        value,
+      ]) => {
+        if ([
+          null,
+          "",
+        ].includes(value)) delete data[key];
+      });
+    }
+
+    $.export("$summary", `Successfully created job ${data.id}`);
```
Comment (lcaresia) [inline components/detrack/actions/create-job/create-job.ts:68]:
```suggestion
    $.export("$summary", `Successfully created job with ID ${data.id}`);
```

File: components/detrack/app/detrack.app.ts [hunk 2]
```diff
@@ -1,13 +1,36 @@
...
+    async _httpRequest({
+      $ = this,
+      endpoint,
+      ...args
+    }: HttpRequestParams): Promise<object> {
+      return axios($, {
+        url: this._baseUrl() + endpoint,
+        headers: {
+          "Content-Type": "application/json",
+          "X-API-KEY": this.$auth.api_key,
```
Comment (lcaresia) [inline components/detrack/app/detrack.app.ts:23]: Move this to a getter method

```suggestion
          "X-API-KEY": this._apiKey(),
```

**Review by lcaresia — CHANGES_REQUESTED** [review]
Hello, just is needed one fix!

File: components/detrack/app/detrack.app.ts (hunk 2 above)
Comment (GTFalcao) [reply components/detrack/app/detrack.app.ts:23]: Applied this

File: components/detrack/actions/create-job/create-job.ts (hunk 1 above)
Comment (GTFalcao) [reply components/detrack/actions/create-job/create-job.ts:68]: Applied this

**Review by lcaresia — APPROVED** [review]
LGTM!

**Comment by DilanAthukorala** [comment]
Verification completed, ready for release.

**Pull request merged**